### DISCUSSION TITLE
Really need help with scripts

### DISCUSSION THREAD

**Discussion by Grinchcreator** [swup/swup#547] [Help] [answered]
So I put swup. the animation works but the scipts don't work on the loaded page. I know people asked about it several times but I don't understand what's wrong with my code.
So here's the link https://github.com/Grinchcreator/Taylor-Swift-Audio-Player-first-project- .
if we open the idex.html it looks fine https://grinchcreator.github.io/Taylor-Swift-Audio-Player-first-project-/index.html the scipts work well. If we go to the next song and press the forward button it freezes (the play/pause button + volume and time sliders
I put the ScriptsPlugin as well but now the animation doesn't work at all

<script src= "script.js"> </script>
 <script src= "scriptVol.js"> </script>
<script src="swup.js"></script>
<script src="SwupScriptsPlugin.min.js"></script>
 <script> const swup = new Swup({
  plugins: [new SwupScriptsPlugin(  body: true
)]
If I delete the plugin it freezes..

I guess the problem is deeper. Please help

**Comment by daun** [comment]
The swup container `<main id="swup">` has to be present on every page that is loaded, otherwise swup will abort the transition. There's a few pages on your site that have a different structure. Also, I had an error about `transition-*` classes missing, so double-check you have those everywhere.

Regarding the scripts, have you tried the suggestions in the docs about [Reloading Javascript](https://swup.js.org/getting-started/reloading-javascript)? I'd strongly suggest doing that first. If you did that and are still having trouble, let us know.

**Comment by Grinchcreator** [comment]
Yes I did it. The `<main id="swup"> `wasn't the issue. The scripts for audio player (for the button and sliders) are still not working after each transition. They work only if we refresh the page.

![image](https://user-images.githubusercontent.com/119819551/207136804-008f2ddf-dfa3-4554-a8fd-4249558ece42.png)

I dunno.... maybe I should add some one time self refresh for each page?

**Comment by daun** [comment, marked as answer]
I think you should read the linked docs page again. There's example code for your exact use case. You'll need to listen for swup's `contentReplaced` event and initialize any new components in the new content whenever that event fires.
